### ISSUE TITLE
Won't paste into payment fields on boostmobile.com

### ISSUE THREAD

**Issue by miraclemandi** [jswanner/DontFuckWithPaste#17]
Hello. I have appreciated your extension on many sites but for some reason I cannot get it to work when trying to paste my debit card info into any of the payment fields on the boostmobile site. I tried deleting all of the 'exclude' addresses in options & it still doesn't work. I don't know anything about java script regular expressions so maybe I just don't know the proper thing to add to the 'include' section in options. 

Thank you for your time & help and of course the extension. Blessings to you & yours! Amanda

![boost mobile payment fields](https://cloud.githubusercontent.com/assets/23058787/19694519/a5a5eb3e-9aad-11e6-9cb4-0a74fe79b8b1.jpg)

**Comment by jswanner** [comment]
**TL;DR**:

It sounds like this site isn't blocking the `paste` event, instead is doing something else to prevent pasting of text. Try right-clicking on the form field, and then select "Paste" from there.

**Longer version**:

When you attempt to paste text into a text box in a browser, there's a specific event called `paste` thats triggered, which allows the creators of websites to hook into those events if they so choose. This extension essentially prevents the website's JavaScript code from receiving those `paste` events, that's because a common reason for a website to hook into the `paste` event is to block it. Instead the extension hooks into the event and tells the browser to perform it's normal action.

Unfortunately, there are other ways that website creators can use JavaScript to interfere with how users interact with their site; there are other events that they can hook into that could effectively block pasting of text. And, there's no extension that can fix all the ways that a website creator can interfere with user interaction _and_ also keep the website in question working properly.

It sounds like this site isn't blocking the `paste` event, instead is doing something else to prevent pasting of text. Try right-clicking on the form field, and then select "Paste" from there. Even if you can paste the text in, there's no guarantee that the site will behave normally.

I recently made a car rental reservation with https://www.avis.com, and I was able to paste my credit card number into their form field, but whenever I submitted the form, it always claimed "Invalid card number." It wasn't until I typed out the credit card number digit-by-digit that it accepted my credit card number as valid. I didn't go diving into their code to see what they were doing exactly, but it seems they were hooking into one of the key press events, and as each number was pressed they were storing that number somewhere else in addition to the form field itself. Then, when the form was submitted, they then used the data from this other location to validate the card number. Which is a really weird/stupid way of doing things, but that's what their programmers have done, and there's no way for me to fix their code as an extension author. Anyway, this is a long story to just say that there's no browser extension that can fix a poorly built website.

**Comment by miraclemandi** [comment]
Ty I appreciate your response. I've tried right-clicking and keyboard commands to no avail. Until recently I could always click on my clipboard manager and select 'paste to active window' and get around it that way but not anymore. 

Could you tell me the proper javascript regular expression to type to add the boostmobile site to the 'Include URL Patterns' list? I left the default one: */ and added: */https://boostmobile.com. Should I try to add "https://myaccount.boostmobile.com/" since that is the actual url where I actually enter my payment information?

Thank you again. Be blessed! Amanda

**Comment by jswanner** [comment]
If you still have the default value (`.*`) in the "Include URL Patterns", then you just want to make sure you don't have anything that would match the URL in the "Exclude URL Patterns". So, you don't want `.*` nor anything with `boostmobile` as an exclude pattern.

I don't have a boostmobile account, so I can't get to the page where you are having trouble, but it seems like they are being very controlling of user interaction on their website via JavaScript, and this extension is not going to be able to fix that.

**Comment by miraclemandi** [comment]
Ok thank you. I did try it and they do have it blocked. This being the only site so far that I've had this issue with is probably not going to kill me. I will just have to type it out. I appreciate your help. Be blessed! Amanda
